api/service.go

```diff
@@ -6,182 +6,208 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/google/uuid"
+	logging "github.com/ipfs/go-log"
 	ma "github.com/multiformats/go-multiaddr"
 	pb "github.com/textileio/go-threads/api/pb"
-	core "github.com/textileio/go-threads/core/db"
-	"github.com/textileio/go-threads/crypto/symmetric"
+	coredb "github.com/textileio/go-threads/core/db"
+	core "github.com/textileio/go-threads/core/service"
+	"github.com/textileio/go-threads/core/thread"
+	sym "github.com/textileio/go-threads/crypto/symmetric"
 	"github.com/textileio/go-threads/db"
+	"github.com/textileio/go-threads/util"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-// service is a gRPC service for a db manager.
-type service struct {
+var (
+	log = logging.Logger("threadsapi")
+)
+
+// Service is a gRPC service for a DB manager.
+type Service struct {
 	manager *db.Manager
 }
 
-// NewDB adds a new db into the manager.
-func (s *service) NewDB(context.Context, *pb.NewDBRequest) (*pb.NewDBReply, error) {
-	log.Debugf("received new db request")
+// Config specifies server settings.
+type Config struct {
+	RepoPath string
+	Debug    bool
+}
+
+// NewService starts and returns a new service with the given threadservice.
+// The threadservice is *not* managed by the server.
```

Comment:
After this is merged, we should make the "service" (threadservice) > "network" refactor. Lots of ambiguous working in this comment and the method signature just below.

Reply:
Totally. I can smash that in after this quick.